tensorflow_datasets/scripts/random_image_generator.py

```diff
@@ -0,0 +1,132 @@
+
+""" Generate random images that compress better.
+    By replacing the larger images with more compressable equivalents. """
+
+import os
+import tempfile
+import zipfile
+import tarfile
+from PIL import Image
+import numpy as np
+from absl import app
+from absl import flags
+
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string("dir_path",
+                    "testing/test_data/fake_examples",
+                    "path to the directory which contains files")
+
+def image_process(filepath):
+    """ Generate random images and remove noise of the
+        images to compress better.
+
+          Args:
+           filepath: path of the images to get processed
+
+    """
+    image = np.array(Image.open(filepath))
+    grey = int(hash(filepath) % 255)
+    image = np.ones_like(image)*grey
+    image = Image.fromarray(image)
+    image = image.convert('RGB')
```

Comment:
No actually i forgot to remove this.

Reply:
Actually if i remove this then it shows an error that image is not converting into jpeg.

Reply:
cannot write mode RGBA as JPEG

Reply:
This is because some images contains alpha channel.

Reply:
You should save the images the same way they were. If the images has 4 channels, you should save it as 4 channels, if the images has 1 channel, you should save it with 1 channel.

Reply:
The question is how loading a jpeg channel load 4 channels. Something feels wrong to me.

Reply:
Ok so shall i remove this or not.

Reply:
Because pillow JPEG doesn't allow RGBA. Now what should i do

Reply:
You should try to investigate why `np.array(Image.open('x.jpeg'))` shape is `(h, w, 4)`. For instance, for which path does it happen ? This might indicates that one of the images is a png saved as jpeg.

The script should save images the same way they are loaded, so if `x.jpeg` had 4 channels, then the saved images should have 4 channels too.

Just to get a first overview, could you send a PR of the fake_data dir modified with your script (without this fix) so I can get an overview of if the overall script is working or not ?

Reply:
Sure

Reply:
I have sended you a new PR. Please have a look.